### PR TITLE
feat: move previous attrbute location

### PR DESCRIPTION
**Pull request by christojeffrey**
move 
- link.liftedSource: GraphDataNode 
- link.liftedTarget: GraphDataNode

- node.outgoingLinkLifted: GraphDataEdge
- node.incomingLinkLifted: GraphDataEdge

to be created when creating graphData

### CODE REVIEW

**Comment by christojeffrey** [comment]
IMPORTANT: merge [this](https://github.com/christojeffrey/software-visualization/pull/36) first

**Comment by christojeffrey** [comment]
reasoning behind this PR already confirmed from group chat

**Comment by christojeffrey** [comment]
IMPORTANT: merge [this](https://github.com/christojeffrey/software-visualization/pull/37) first
PR 33 (this one) is used to **add** the attribute.

PR 37 (link above) is used to **remove** previous way of deriving the attribute

File: svelte/src/scripts/create-graph-data/index.ts [hunk 1]
```diff
@@ -21,17 +21,17 @@ function assignParentReference(nodes: GraphDataNode[]) {
 	});
 }
 
-function assignOutgoingAndIncomingLinksAndOriginalSourceAndTargetReference(
+function assignOutgoingIncomingLinksAndOriginalLiftedSourceTargetReference(
```
Comment (Perry26) [inline svelte/src/scripts/create-graph-data/index.ts:24]: What a name.

You can just call it something like assignLinkReferences and use a jsdoc comment to explain what exectly it does.

File: svelte/src/scripts/create-graph-data/index.ts [hunk 2]
```diff
@@ -69,25 +96,36 @@ export function createGraphData(convertedData: ConvertedData): GraphData {
...
-	const graphDataLinks = assignOutgoingAndIncomingLinksAndOriginalSourceAndTargetReference(
+	const graphDataLinks = assignOutgoingIncomingLinksAndOriginalLiftedSourceTargetReference(
 		links,
 		flattenNodes,
 		graphDataFlattenNodes
 	);
 
+	// create nodesDict
+	const nodesDict: { [id: string]: GraphDataNode } = {};
+	graphDataFlattenNodes.forEach((node) => {
```
Comment (Perry26) [inline svelte/src/scripts/create-graph-data/index.ts:117]: Since Satrio took issue with the flattenNodes and the nodesDict specifically, maybe the GraphData should be wrapped in a class? Then those interfaces could be a façade. Though that may be for another PR.

**Review by Perry26 — APPROVED** [review]
Since the other pr is approved, we can merge this as well.

File: svelte/src/scripts/create-graph-data/index.ts (hunk 1 above)
Comment (christojeffrey) [reply svelte/src/scripts/create-graph-data/index.ts:24]: agree :v
